### PR TITLE
Added logs

### LINKED ISSUE #65: improve logging
as a developer i want more logs to be generated by the base task-api library so i can more easily debug my own written apps.

- add log-level as argument and in docs
- print warnings and errors where applicable
- print info on rpc request and response
- print debug when extended data is needed

by default the app will run on INFO like golem
when task-api-dev is passed to golemapp it should run in DEBUG mode ( issue TBC and linked here)

### PR DESCRIPTION
**Pull request by maaktweluit**
in all ( non util ) files and an argument to entrypoint.py to set the log level

resolves #65

### CODE REVIEW

File: python/golem_task_api/client.py [hunk 1]
```diff
@@ -63,10 +66,15 @@
         try:
             response = await client.Check(request)
             if response.status == HealthCheckResponse.SERVING:
+                logger.debug('Service health check completed with success')
                 return channel
         except (StreamTerminatedError, ConnectionError):
             pass
         channel.close()
+        logger.debug(
+            'Service health check failed, will try again in 100ms. deadline=%r',
```
Comment (jiivan) [inline python/golem_task_api/client.py]: Maybe use a common variable for the delay - (100ms and asyncio.sleep(0.1))

File: python/golem_task_api/entrypoint.py [hunk 2]
```diff
@@ -21,7 +25,16 @@
         requestor_lifecycle_handler: Optional[AppLifecycleHandler] = None,
         provider_handler: Optional[ProviderAppHandler] = None,
         provider_lifecycle_handler: Optional[AppLifecycleHandler] = None,
+        log_level: str = None,
 ):
+    level = log_level or logging.INFO
+    logging.basicConfig()
```
Comment (jiivan) [inline python/golem_task_api/entrypoint.py]: I know it's a starting point, but I haven't seen a case when it wasn't quickly changed into `dictConfig()` :) So IMHO it's better to implement it in the beginning.

File: python/golem_task_api/client.py (hunk 1 above)
Comment (maaktweluit) [reply python/golem_task_api/client.py]: good one! added the var, did print a float to make it more the same

File: python/golem_task_api/entrypoint.py (hunk 2 above)
Comment (maaktweluit) [reply python/golem_task_api/entrypoint.py]: great to prepare for the future, updated!

**Pull request merged**